### PR TITLE
Studied section 10 - interfaces.

### PR DESCRIPTION
**Pull request by Yurii666**
studied section 10 - interfaces.
printed examples of use.
some examples don't work.

### CODE REVIEW

File: src/java_10thEdition/AskMe.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                System.out.println("Later");
+                break;
+            case SOON:
+                System.out.println("Soon");
+                break;
+            case NEWER:
+                System.out.println("newer");
+                break;
+        }
+    }
```
Comment (Yurii666) [inline src/java_10thEdition/AskMe.java]: how the ansver method in the AskMe class does here I did not understand

File: src/java_10thEdition/AskMe.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    int MAYBE = 2;
+    int LATER = 3;
+    int SOON = 4;
+    int NEWER = 5;
+}
+class Question implements SharedConstants{
+    Random rand = new Random();
+    int ask(){
+        int prod = (int)(100* rand.nextDouble());
+        if(prod<30)
```
Comment (Yurii-Loienko) [inline src/java_10thEdition/AskMe.java]: brackets

File: src/java_10thEdition/AskMe.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package java_10thEdition;
+
+import java.sql.SQLOutput;
+import java.util.Random;
+
+interface SharedConstants{
+    int NO = 0;
```
Comment (Yurii-Loienko) [inline src/java_10thEdition/AskMe.java:6]: Can you make it private?

File: src/java_10thEdition/AskMe.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    int YES = 1;
+    int MAYBE = 2;
+    int LATER = 3;
+    int SOON = 4;
+    int NEWER = 5;
+}
+class Question implements SharedConstants{
+    Random rand = new Random();
+    int ask(){
+        int prod = (int)(100* rand.nextDouble());
```
Comment (Yurii-Loienko) [inline src/java_10thEdition/AskMe.java]: Can you use nextInt here?

File: src/java_10thEdition/AskMe.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        else if (prod<75)
+            return LATER;
+        else if (prod<98)
+            return SOON;
+        else return NEWER;
+    }
+}
+ class AskMe implements SharedConstants{
+    static void ansver(int result){
+        switch (result) {
```
Comment (Yurii-Loienko) [inline src/java_10thEdition/AskMe.java:36]: What happens if the result value doesn't have any of provided cases? For example result = 100000
Your method should be versatile

File: src/java_10thEdition/IFTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            return 0;
+        }else
+            return  stck[tos--];
+    }
+}
+public class IFTest {
+    public static void main(String[] args) {
+        FixedStack mystack1 = new FixedStack(5);
+        FixedStack mystack2 = new FixedStack(8);
+        for(int i=0;i<5;i++) mystack1.push(i);
```
Comment (Yurii-Loienko) [inline src/java_10thEdition/IFTest.java]: brackets

**Review by Yurii-Loienko — CHANGES_REQUESTED** [review]

File: src/java_10thEdition/AskMe.java (hunk 3 above)
Comment (Yurii666) [reply src/java_10thEdition/AskMe.java:6]: No. It is interface.

File: src/java_10thEdition/AskMe.java (hunk 4 above)
Comment (Yurii666) [reply src/java_10thEdition/AskMe.java]: yes, I can do it by changing the code
int prod = (int)(rand.nextInt(100));

File: src/java_10thEdition/AskMe.java (hunk 2 above)
Comment (Yurii666) [reply src/java_10thEdition/AskMe.java]: I can insert them in the future everywhere, in all cycles, if you want.
in this case, in this package I have all the programs that I print from the book for a better understanding of what is happening in them, if you want, I will add parentheses here too

File: src/java_10thEdition/AskMe.java (hunk 5 above)
Comment (Yurii666) [reply src/java_10thEdition/AskMe.java:36]: if I understood this program correctly, (I did not understand the Ansver method works at all) the next line
--int prod = (int)(rand.nextInt(100));--
in class Question transmits numbers from 0 to 100 maximum, 100000 cannot be here.
but it is not exactly

Comment (Loienko) [reply src/java_10thEdition/AskMe.java:36]: Do you see my comment?

File: src/java_10thEdition/AskMe.java (hunk 2 above)
Comment (Yurii-Loienko) [reply src/java_10thEdition/AskMe.java]: It is not only my wish. this is a condition of clean code

Comment (Yurii-Loienko) [reply src/java_10thEdition/AskMe.java]: You should do it in all your future tasks

File: src/java_10thEdition/AskMe.java (hunk 5 above)
Comment (Yurii-Loienko) [reply src/java_10thEdition/AskMe.java:36]: Yes, you are right.
But let's imagine that someone else will pass such a value to the method.
You just need to add default case

File: src/java_10thEdition/AskMe.java (hunk 1 above)
Comment (Yurii-Loienko) [reply src/java_10thEdition/AskMe.java]: You pass the int value as result. 
Please, read the article to understand how to work switch/case

Comment (Yurii666) [reply src/java_10thEdition/AskMe.java]: I do not understand why the entire ansver method is launched if in the main method we create an object of the Question class, use the variables of the Question class, and nowhere I see that we run and use the objects of the ansver method. this is not clear to me

Comment (Yurii-Loienko) [reply src/java_10thEdition/AskMe.java]: Let's imagine that I created one new method which will use your method answer. And my method will be have one more case like ASAP(as soon as possible) or something else. But I will break the main rules of the Open close principle. So I need to create one new method which will be the same as you have already written but with only one other case. 
Method two: Imagine that someone else passed your method a different value that is not present in your implementation.
Switch case implementation should have 'default' cases by default like
switch ('Test'):{
case 'Test':
 bla, bla,
 break;
default:
 break;
 https://docs.oracle.com/javase/tutorial/java/nutsandbolts/switch.html

File: src/java_10thEdition/AskMe.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    int YES = 1;
+    int MAYBE = 2;
+    int LATER = 3;
+    int SOON = 4;
+    int NEWER = 5;
+}
+class Question implements SharedConstants{
+    Random rand = new Random();
+    int ask(){
+        int prod = (int)(rand.nextInt(100));
```
Comment (Yurii-Loienko) [inline src/java_10thEdition/AskMe.java]: In the previous version you used nextDouble() and get 'Double' value. After that you use type casting to int.
Could you explain the difference between double and int? What does typecasting mean?

**Review by Yurii-Loienko — CHANGES_REQUESTED** [review]

File: src/java_10thEdition/AskMe.java (hunk 1 above)
Comment (Yurii666) [reply src/java_10thEdition/AskMe.java]: added default in AskMe

File: src/java_10thEdition/AskMe.java (hunk 7 above)
Comment (Yurii666) [reply src/java_10thEdition/AskMe.java]: double is a floating point numeric value. int is an integer numeric value. type casting is changing the type of a variable and fitting a value to a new data type. when moving from double to int, the numbers after thepoint are removed and the number is rounded to an integer.
I agree, there is no need for type casting

**Comment by Yurii-Loienko** [comment]
Looks good for me

**Review by Yurii-Loienko — APPROVED** [review]

**Pull request merged**